posts/how-to-commit-code-with-the-git-command-line-interface.md

```diff
@@ -0,0 +1,210 @@
+
+Suppose that we’re committing changes to update the readme file and add a getting-started file. Our commit message could be:
+
+> Update readme and add getting-started file
+
+We’ll use this message with the `git commit` command in the next step.
+
+### Committing
+
+Having a commit message in mind, we’re now ready to run the `git commit` command.
+
+We specify the commit message by providing it as an argument to `git commit`. The message should be a string in quotes preceded by the `-m` flag, which stands for “message.”
+
+With our example commit message, the command would look like this:
+
+```sh
+git commit -m “Update readme and add getting-started file”
+```
+
+Here’s the output when we run this command:
+
+```
+[main 5388e04d38] Update readme and add getting-started file
+ 2 files changed, 1 insertion(+), 1 deletion(-)
+ create mode 100644 getting-started.txt
+```
+
+The above output tells us a few things about the commit that we just created: the number of files changed, the number of insertions and deletions (which is a count of lines affected), and the list of created files.
+
+Because `git commit` acts on our local file system and doesn’t automatically send our commit anywhere, we’ll need to use the `git push` command to push our commit to a remote repository such as GitHub.
+
+## Step 5 — Checking the commit
+
+After running `git commit`, we can check that it was created successfully with the `git log` command.
+
+The `git log` command outputs a list of recent commits. Since we’re only interested in checking the last commit, we can shorten the output by adding the `-1` option, which will display only the most recent entry.
+
+```sh
+git log -1
+```
+
+This helps us confirm that the last commit is, in fact, the one that we just created. Here’s the result when we run `git log -1` after creating our commit:
+
+```
+commit 5388e04d38e5de13e3968f8d5e1932a9b41e5e53 (HEAD -> main)
+Author: Marek <marek@sourcegraph.com>
+Date:   Tue Aug 31 18:47:11 2021 -0400
+
+    Update readme and add getting started file
+```
+
+In the above output of `git log -1`, we see the commit ID and the branch name (`main`) on the first line. The output also includes the author and date, followed by the commit message.
+
+Based on this output, we can conclude that the commit was successful and is now part of the repository.
+
+## Learn more
+
+In this tutorial, we explored how to create Git commits with command-line tools. There are other ways to create Git commits, and there are many tools that can facilitate the task.
+
+Check out our tutorials about committing code to Git with other tools:
+
+- [How to commit code to a Git repository with Visual Studio Code](/how-to-commit-code-to-a-git-repository-with-visual-studio-code)
```

Comment:
These both need to be live or commented out before this can be merged as we can't have broken links in prod

Reply:
I commented out these links and will uncomment them when they are live.